src/netex_idf/lines.rs

```diff
@@ -158,29 +169,63 @@ fn make_networks_companies(
     Ok((networks, companies, map_line_network))
 }
 
-pub fn from_path(path: &std::path::Path, collections: &mut Collections) -> Result<()> {
-    info!("Reading {:?}", path);
+fn make_physical_and_commercial_modes(
+    lines_netex_idf: &CollectionWithId<LineNetexIDF>,
+) -> Result<(
+    CollectionWithId<PhysicalMode>,
+    CollectionWithId<CommercialMode>,
+)> {
+    let mut physical_modes = CollectionWithId::default();
+    let mut commercial_modes = CollectionWithId::default();
+    let modes: BTreeSet<_> = lines_netex_idf.values().map(|l| &l.mode).collect();
+    for m in modes {
+        let (physical_mode_id, physical_mode_name, commercial_mode_id, commercial_mode_name) =
+            skip_fail!(MODES
+                .get::<str>(&m)
+                .map(|m| {
+                    let (pi, pn) = m.physical_mode;
+                    let (ci, cn) = m.commercial_mode;
+                    (pi, pn, ci, cn)
+                })
+                .ok_or_else(|| format_err!("{} not found", m)));
+        physical_modes.push(PhysicalMode {
+            id: physical_mode_id.to_string(),
+            name: physical_mode_name.to_string(),
+            ..Default::default()
+        })?;
+        commercial_modes.push(CommercialMode {
+            id: commercial_mode_id.to_string(),
+            name: commercial_mode_name.to_string(),
+        })?;
+    }
+    Ok((physical_modes, commercial_modes))
+}
 
+pub fn from_path(
+    path: &std::path::Path,
+    collections: &mut Collections,
+) -> Result<CollectionWithId<LineNetexIDF>> {
+    info!("Reading {:?}", path);
     let mut file = File::open(&path).with_context(ctx_from_path!(path))?;
     let mut file_content = String::new();
     file.read_to_string(&mut file_content)?;
 
-    if let Ok(elem) = file_content.parse::<Element>() {
+    let lines_netex_idf = if let Ok(elem) = file_content.parse::<Element>() {
```

Comment:
This is a big `if/else` block. It might be interesting to extract a function here.

Reply:
This code might change in the future since changes are expected on Netex specification. So don't worry about that, we'll take care of refactoring later.